### PR TITLE
 QMM 4.1.3

### PR DESCRIPTION
**Pull request by MrPurple6411**
- Fix for the Dialog popup as coroutines were being killed before the main menu shows up in BZ.
- Log file filter patch #227

### CODE REVIEW

File: QModPluginEmulator/QModPluginGenerator.cs [hunk 1]
```diff
@@ -108,6 +108,13 @@ private static IEnumerator InitializeQMM(IEnumerator result)
 
                 SummaryLogger.ReportIssues(ModsToLoad);
                 SummaryLogger.LogSummaries(ModsToLoad);
+                if(Patcher.Dialogs.Count > 0)
```
Comment (PrimeSonic) [inline QModPluginEmulator/QModPluginGenerator.cs:111]: No need to check `Count` when using a `foreach`

File: QModManager/Utility/Dialog.cs [hunk 2]
```diff
@@ -71,47 +71,70 @@ internal enum DialogColor
...
+                coroutineHandler.AddComponent<DummyBehaviour>().StartCoroutine(ShowDialogEnumerator());
+            }
+            else if(coroutineHandler.TryGetComponent(out DummyBehaviour dummyBehaviour))
+            {
+                dummyBehaviour.StartCoroutine(ShowDialogEnumerator());
+            }
+            else
+            {
+                coroutineHandler.AddComponent<DummyBehaviour>().StartCoroutine(ShowDialogEnumerator());
+            }
```
Comment (PrimeSonic) [inline QModManager/Utility/Dialog.cs:91]: Could this work instead? _replacing lines 79-91_
```CS
if(coroutineHandler is null)
{
    coroutineHandler = new GameObject("QModManager Dialog Coroutine");
}
var dummyBehaviour = coroutineHandler.EnsureComponent<DummyBehaviour>();
dummyBehaviour.StartCoroutine(ShowDialogEnumerator());
```

File: QModManager/Utility/Dialog.cs [hunk 3]
```diff
@@ -121,61 +144,54 @@ private IEnumerator ShowDialogEnumerator()
             }
 
             // Turn the dialog blue if the blue parameter is true
-            Sprite sprite = confirmation.gameObject.GetComponentInChildren<Image>().sprite;
-            if (color == DialogColor.Blue)
+            Sprite = confirmation.gameObject.GetComponentInChildren<Image>().sprite;
```
Comment (PrimeSonic) [inline QModManager/Utility/Dialog.cs:147]: local variables should start lowercase

Comment (MrPurple6411) [reply QModManager/Utility/Dialog.cs:147]: it is no longer local as it gets used in the callback which moved out of that method

File: QModManager/Utility/Dialog.cs (hunk 2 above)
Comment (MrPurple6411) [reply QModManager/Utility/Dialog.cs:91]: yea it should.

File: QModPluginEmulator/QModPluginGenerator.cs (hunk 1 above)
Comment (MrPurple6411) [reply QModPluginEmulator/QModPluginGenerator.cs]: no clue why I did that lol 
bb148d0

**Review by PrimeSonic — APPROVED** [review]

**Comment by PrimeSonic** [comment]
Remember to re-build

**Pull request merged**